### PR TITLE
Add some main and command-line args

### PR DESCRIPTION
**Pull request by kirkwood**
First piece of the release puzzle: a baby main with simple command-line arguments. I want to get this out for commentary before I do more on it.

Right now, all it does is allow the user to pick a port (which defaults to 30304) and to specify a "home" node for bootstrapping the network.

This PR includes some sbt support for bundling the jar with `sbt assembly`, and it can be run by naming the resulting fat jar on the command line. For example:

```bash
$ sbt assembly
[info] Loading settings from plugins.sbt ...
[info] Loading global plugins from /home/kirkwood/.sbt/1.0/plugins
...
[info] DistanceSpec:
[info] A PeerNode of width n bytes
[info] - should have distance to itself equal to 8n
[info] A node with key all zeroes (0000000000000000000000000000000000000000000000000000000000000000)
[info] - should compute distance correctly
[info] A node with key all ones (ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff)
[info] - should compute distance correctly
...
[info] Tests: succeeded 20, failed 0, canceled 0, ignored 0, pending 0
[info] All tests passed.
[info] Checking every *.class/*.jar file's SHA-1.
...
[info] Packaging /home/kirkwood/src/crypto/rchain/rchain/comm/target/scala-2.12/comm-assembly-0.1-SNAPSHOT.jar ...
[info] Done packaging.
[success] Total time: 6 s, completed Nov 30, 2017 6:00:06 PM

$ java -jar target/scala-2.12/comm-assembly-0.1-SNAPSHOT.jar -p 12345
09:05:51.774 [main] INFO main - Listening for traffic on #{Network rnode://64af30bbbd9045e48c13627cc2963d52@localhost:12345}.
```

Yikes!

The current dependency list includes these jars:
 * scala-library-2.12.3.jar
 * scallop_2.12-3.0.3.jar
 * scalapb-runtime_2.12-0.6.0.jar
 * scala-uri_2.12-0.5.0.jar
 * parboiled_2.12-2.1.4.jar
 * shapeless_2.12-2.3.2.jar
 * lenses_2.12-0.4.12.jar
 * protobuf-java-3.3.1.ja

CORE-100 CORE-95 #comment First pass at programmification.

### CODE REVIEW

**Comment by kirkwood** [comment]
Ahhh...that's what I was hoping someone would tell me. Works like a charm; will update PR.

**Comment by kirkwood** [comment]
@DoSOfRedRiver did the things: there should be logging and fat jar support. Thanks for that.

File: comm/src/main/scala/coop/rchain/comm/main.scala [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+object Main {
+  def main(args: Array[String]): Unit = {
+    val logger = scalalogging.Logger("main")
+    val conf = Conf(args)
+    val name = UUID.randomUUID.toString.replaceAll("-", "")
+    val addy = s"rnode://$name@localhost:${conf.port()}"
+    val net = p2p.Network(addy)
+    logger.info(s"Listening for traffic on $net.")
+
+    conf.home.foreach { address =>
```
Comment (DoSOfRedRiver) [inline comm/src/main/scala/coop/rchain/comm/main.scala]: I guess that if no "home" were specified application will close without any message.

File: comm/build.sbt [hunk 2]
```diff
@@ -5,16 +5,28 @@ PB.targets in Compile := Seq(
   scalapb.gen(javaConversions = true) -> (sourceManaged in Compile).value
 )
 
+mainClass in (Compile, run) := Some("coop.rchain.comm.Main")
+mainClass in (Compile, packageBin) := Some("coop.rchain.comm.Main")
```
Comment (DoSOfRedRiver) [inline comm/build.sbt]: Does this work for fat jar? I think there is should be definition of main class for "assembly". You can take a look at roscala's build.sbt

File: comm/src/main/scala/coop/rchain/comm/main.scala (hunk 1 above)
Comment (kirkwood) [reply comm/src/main/scala/coop/rchain/comm/main.scala]: I think for a general rchain client/node, you're probably right. I _do_ need a way to get this off the ground, though.

File: comm/build.sbt (hunk 2 above)
Comment (kirkwood) [reply comm/build.sbt]: It does work, but I've simplified it to exclude the above and only include assembly as you suggest.

**Pull request merged**